kinesis/src/main/scala/akka/stream/alpakka/kinesis/KinesisSchedulerSettings.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.kinesis
+
+import scala.concurrent.duration._
+
+final class KinesisSchedulerSourceSettings(val bufferSize: Int, val backpressureTimeout: FiniteDuration) {
+  require(
+    bufferSize >= 1,
+    "Buffer size must be greater than 0; use size 1 to disable stage buffering"
+  )
+}
+final class KinesisSchedulerCheckpointSettings(val maxBatchSize: Int, val maxBatchWait: FiniteDuration) {
+  require(
+    maxBatchSize >= 1,
+    "Batch size must be greater than 0; use size 1 to commit records one at a time"
+  )
+}
+
+object KinesisSchedulerSourceSettings {
+
+  val defaultInstance = new KinesisSchedulerSourceSettings(1000, 1.minute)
```

Comment:
Return the default from parameterless `apply` and `create` methods.

Reply:
Should I add a `create` method in each companion obj here?